src/main/java/frc/robot/subsystems/endEffector/factory/EndEffectorSparkMaxBuilder.java

```diff
@@ -0,0 +1,103 @@
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.IDs;
+import frc.robot.Robot;
+import frc.robot.hardware.digitalinput.IDigitalInput;
+import frc.robot.hardware.digitalinput.chooser.ChooserDigitalInput;
+import frc.robot.hardware.digitalinput.supplied.SuppliedDigitalInput;
+import frc.robot.hardware.mechanisms.wpilib.SimpleMotorSimulation;
+import frc.robot.hardware.rev.motors.*;
+import frc.robot.hardware.signal.supplied.SuppliedDoubleSignal;
+import frc.robot.subsystems.endEffector.EndEffector;
+import frc.robot.subsystems.endEffector.EndEffectorConstants;
+
+public class EndEffectorSparkMaxBuilder {
+
+	private static final int NUMBER_OF_MOTORS = 1;
+	private static final boolean IS_INVERTED = false;
+	private static final boolean SET_BRAKE = false;
+	private static final int CURRENT_LIMIT = 30;
+
+	private static final double POSITION_CONVERSION_FACTOR = 1;
+	private static final double MOMENT_OF_INERTIA = 0.001;
+
+	private static final Double DEBOUNCE_TIME_SECONDS = 0.1;
+
+	private enum LimitSwitchDirection {
+
+		FORWARD,
+		REVERSE;
+
+	}
+
+	private static void configMotor(SparkMaxMotor sparkMaxMotor) {
+		SparkMaxConfig config = new SparkMaxConfig();
+		config.inverted(IS_INVERTED);
+		config.smartCurrentLimit(CURRENT_LIMIT);
+
+		config.limitSwitch.forwardLimitSwitchEnabled(false);
+		config.limitSwitch.forwardLimitSwitchType(LimitSwitchConfig.Type.kNormallyOpen);
+
+		config.limitSwitch.reverseLimitSwitchEnabled(false);
+		config.limitSwitch.reverseLimitSwitchType(LimitSwitchConfig.Type.kNormallyOpen);
+
+		sparkMaxMotor.applyConfiguration(new SparkMaxConfiguration().withSparkMaxConfig(config));
+		sparkMaxMotor.setBrake(SET_BRAKE);
+	}
+
+	private static BrushlessSparkMAXMotor generateMotor(String logPath, SparkMaxWrapper sparkMaxWrapper) {
+		SimpleMotorSimulation simulation = new SimpleMotorSimulation(
+			new DCMotorSim(
+				LinearSystemId.createDCMotorSystem(DCMotor.getNEO(NUMBER_OF_MOTORS), MOMENT_OF_INERTIA, 1 / POSITION_CONVERSION_FACTOR),
+				DCMotor.getNEO(NUMBER_OF_MOTORS)
+			)
+		);
+
+		BrushlessSparkMAXMotor motor = new BrushlessSparkMAXMotor(logPath, sparkMaxWrapper, simulation, new SysIdRoutine.Config());
+		configMotor(motor);
+		return motor;
+	}
+
+	private static IDigitalInput generateBeamBreaker(SparkMaxWrapper sparkMaxWrapper, String logPath, LimitSwitchDirection limitSwitch) {
+		if (Robot.ROBOT_TYPE.isReal()) {
+			return switch (limitSwitch) {
+				case FORWARD ->
+					new SuppliedDigitalInput(() -> sparkMaxWrapper.getForwardLimitSwitch().isPressed(), new Debouncer(DEBOUNCE_TIME_SECONDS));
+				case REVERSE ->
+					new SuppliedDigitalInput(() -> sparkMaxWrapper.getReverseLimitSwitch().isPressed(), new Debouncer(DEBOUNCE_TIME_SECONDS));
+			};
+		} else {
+			return new ChooserDigitalInput(logPath);
+		}
```

Comment:
```suggestion
		if (!Robot.ROBOT_TYPE.isReal()) {
			return new ChooserDigitalInput(logPath);
		}
		return switch (limitSwitch) {
				case FORWARD ->
					new SuppliedDigitalInput(() -> sparkMaxWrapper.getForwardLimitSwitch().isPressed(), new Debouncer(DEBOUNCE_TIME_SECONDS));
				case REVERSE ->
					new SuppliedDigitalInput(() -> sparkMaxWrapper.getReverseLimitSwitch().isPressed(), new Debouncer(DEBOUNCE_TIME_SECONDS));
			};
		}
```

Personally I think this is cleaner since it "saves" one layer of indenations, improving readability.